Cython/Compiler/PyrexTypes.py

```diff
@@ -4019,6 +4021,100 @@ def check_nullary_constructor(self, pos, msg="stack allocated"):
         if constructor is not None and best_match([], constructor.all_alternatives()) is None:
             error(pos, "C++ class must have a nullary constructor to be %s" % msg)
 
+class CppScopedEnumType(CType):
```

Comment:
I also wonder if some of the duplicated functionality with CEnumType could be shared here. Maybe inheritance?

Reply:
See comment https://github.com/cython/cython/pull/3640/files#r436307155 that I mean to write here instead.

Reply:
In terms of `CIntLike` - you seem to be allowing conversion to/from Python ints (which is probably a reasonable thing to do since it gives a way to pass the values to/from Python). Does that mean that they _are_ int-like from a Cython point-of-view (even if not from a C++ point-of-view)? I'm not 100% sure here to be honest.

Reply:
> Does that mean that they are int-like from a Cython point-of-view (even if not from a C++ point-of-view)?

Yes, precisely. C++ enums aren't *trivially* convertible to integers, but can be casted explicitly to their `underlying_type`.

I think it would be more precise here not to make `CppScopedEnumType` be a `CIntLike`, but rather to explicitly call on the functionality of the `underlying_type` when required -- exactly like we discussed [here](https://github.com/cython/cython/pull/3640/#discussion_r437339735).